x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -867,6 +868,10 @@ private static Expression processBinaryOperator(BinaryOperator<?, ?, ?, ?> o) {
             return childrenChanged ? o.replaceChildren(newChildren) : o;
         }
 
+        private static boolean isSupportedCastingForBinaryComparison(DataType type) {
```

Comment:
`allowedType`

I think we convert the string only to non numerics - if the user wants a numeric, they should declare it as such.
So "1" == 1 should not be allowed.

Reply:
Numeric is removed.